### DISCUSSION TITLE
How to wrap output code when printing an error with a large content in html and LateX?

### DISCUSSION THREAD

**Discussion by luifrancgom** [quarto-dev/quarto-cli#9517] [Q&A] [answered]
### Description

I am writing a book and I want to know if it is possible to wrap output when printing an error with a large content in both html and LateX formats.

### Steps to reproduce

````qmd
---
title: "Untitled"
format: 
  html:
    embed-resources: true
  pdf: default
editor: visual
---

# Section 1

```{r}
#| error: true
x <- c("a", "b")
sample(x = x, size = 3)
```
````

#### html

![html](https://github.com/quarto-dev/quarto-cli/assets/50002526/f559ea54-fa86-4ee8-b397-168d793d7d3e)

#### pdf

![pdf](https://github.com/quarto-dev/quarto-cli/assets/50002526/a4a34526-534b-4762-8618-7b75c5f8b0dc)

### Desire behavior

![desire_behavior png](https://github.com/quarto-dev/quarto-cli/assets/50002526/ae2048d9-e0d4-43e9-af82-146542a7459f)

### What I try but what I want to improve

This can be a solution but what I want is something more automatic, if possible. Any help will be appreciated.

````qmd
---
title: "Untitled"
format: 
  html:
    embed-resources: true
  pdf: default
editor: visual
---

# Section 1

```{r}
#| error: true
x <- c("a", "b")
sample(x = x, size = 3)
```

```{r}
#| eval: false
x <- c("a", "b")
sample(x = x, size = 3)
```

```{r}
#| echo: false
cat("Error in sample.int(length(x), size, replace, prob) : cannot take a sample \nlarger than the population when 'replace = FALSE'")
```
````

### Environment

- OS: Windows 11 Pro
- R: 4.3.3
-  IDE:  RStudio 2023.12.1+402
- Quarto: 1.4.552

**Comment by mcanouil** [comment, marked as answer]
A bit of vocabulary, that's not "responsive" that you are looking for but code output wrapping.

Here are some of the previous discussions on this exact topic:
- https://github.com/quarto-dev/quarto-cli/discussions/6017
- https://github.com/quarto-dev/quarto-cli/discussions/7612
- https://github.com/quarto-dev/quarto-cli/discussions/6083
- https://github.com/quarto-dev/quarto-cli/discussions/5728
- https://github.com/quarto-dev/quarto-cli/discussions/3693
- https://github.com/quarto-dev/quarto-cli/discussions/3632
- https://github.com/quarto-dev/quarto-cli/discussions/4121

**Comment by luifrancgom** [comment]
Thank you for your help and pointing me out the correct concept in relation to **output wrapping**. I edit the post taking into account what you mention. However I have some questions. If I try the following solution:

````qmd
---
title: "Untitled"
format: 
  html:
    embed-resources: true
    code-overflow: wrap
  pdf:
    keep-tex: true
    include-in-header: 
       text: |
         \usepackage{fvextra}
         \DefineVerbatimEnvironment{Highlighting}{Verbatim}{breaklines,commandchars=\\\{\}}
    include-before-body:
      text: |
        \RecustomVerbatimEnvironment{verbatim}{Verbatim}{
          showspaces = false,
          showtabs = false,
          breaksymbolleft={},
          breaklines
          % Note: setting commandchars=\\\{\} here will cause an error 
        }
editor: visual
---

# Section 1

```{r}
#| echo: false
grid::unit(lattice::trellis.par.get()$axis.components$bottom$tck, lattice::lattice.options()$layout.widths$right.padding$units)
```

```{r}
library(tidyverse)
```


```{r}
#| error: true

x <- c("a", "b")
sample(x = x, size = 3)
```
````

1. For **pdf** format I get the following result:

![pdfformat](https://github.com/quarto-dev/quarto-cli/assets/50002526/75dd71ce-49b5-485d-89e2-b7baeafaccf7)

It is possible to get rid of `[1] 1char`?

2. For **html** format I get the following result:

![htmlformat](https://github.com/quarto-dev/quarto-cli/assets/50002526/6ce20e8c-f9ca-4136-8233-4d8dd278752c)

As you can check `code-overflow: wrap` doesn't work. However I tried in the case of a book and it works. Why in a quarto document doesn't work but in a quarto book it works?

**Reply by cderv** [reply]
> It is possible to get rid of [1] 1char?

You have this as a results of your first cell with only `echo: false`. If you don't want the output, use `include: false`
See https://quarto.org/docs/computations/execution-options.html

> code-overflow: wrap

I do think this option only applies to source code part, not the output part of it 
Doc says: https://quarto.org/docs/output-formats/html-code.html#code-overflow
> In some cases the width of source code will overflow the available horizontal display space. By default, this will result in a horizontal scroll bar for the code block. However if you prefer not to have scrollbars you can have the longer lines wrap instead.

So for HTML you could handle using a CSS rules where you need to. For example
````css
div.cell-output pre code {
    white-space: pre-wrap;
}
````

Usually, any customization needs to be done in the targeted format when you are not happy with the default from Quarto

Hope it helps

**Comment by luifrancgom** [comment]
Thank you Mickaël Canouil and Christophe Dervieux for your help. As always you really help the Quarto final user.
